pkg/chunk/chunk_store_utils.go

```diff
@@ -117,22 +119,32 @@ func NewChunkFetcher(cacher cache.Cache, cacheStubs bool, storage Client) (*Fetc
 
 // Stop the ChunkFetcher.
 func (c *Fetcher) Stop() {
-	close(c.decodeRequests)
+	select {
```

Comment:
Shouldn't we still close `decodeRequests` somewhere?

Reply:
We are now using the quit channel to signal the other goroutines to return, which avoids the issue of sending on a closed channel.
Channels are garbage collected when they are no longer used regardless if they were closed or not.